src/main/kotlin/step2/lotto/domain/LottoNumber.kt

```diff
@@ -0,0 +1,32 @@
+package step2.lotto.domain
+
+class LottoNumber private constructor(private val value: Int) : Comparable<LottoNumber> {
+    override fun compareTo(other: LottoNumber): Int = value.compareTo(other.value)
+
+    override fun toString(): String {
+        return "$value"
+    }
+
+    companion object {
+        private const val START_RANGE = 1
+        private const val END_RANGE = 45
+        private const val OUT_OF_LOTTO_NUMBER_RANGE_ERROR_MESSAGE = "로또 범위에 해당하는 숫자가 아닙니다. [%d], [$START_RANGE ~ $END_RANGE] 범위에 숫자만 입력하세요."
+        private val LOTTO_NUMBER_RANGE: IntRange = START_RANGE..END_RANGE
+        private val lottoNumbers: Map<Int, LottoNumber> = init()
+
+        private fun init(): Map<Int, LottoNumber> = mutableMapOf<Int, LottoNumber>().apply {
+            repeat(END_RANGE) {
+                this[it + 1] = LottoNumber(it + 1)
+            }
+        }
```

Comment:
```suggestion
        private fun init(): Map<Int, LottoNumber> = (START_RANGE..END_RANGE)
            .mapIndexed { index, number -> index + 1 to LottoNumber(number) }
            .toMap()
```
위와 같이 작성하면 꼭 mutableMap을 만들지 않아도 됩니다. 🙂  kotlin은 불변과 가변 타입을 구분하고 있고 불변이 default이기 때문에 불필요한 경우 가변 타입이나 var을 사용하는 것을 최소화하는 것이 좋습니다.

Reply:
선언과 동시에 초기화하는 방식이 있었네요!
말씀하신 방법을 적용하여 값을 추가하기 위해 사용된 가변 객체 사용을 개선해보겠습니다 👍